wallet/src/backend.rs

```diff
@@ -225,12 +347,12 @@ impl<'a, Meta: Serialize + DeserializeOwned + Send> WalletBackend<'a, CapeLedger
             Some(ret) => Ok((ret, ())),
             None => {
                 let ret = self
-                    .mock_eqs
-                    .lock()
-                    .await
-                    .network()
-                    .nullifier_spent(nullifier);
-                nullifiers.insert(nullifier, ret);
+                    .get_eqs(format!("check_nullifier/{}", nullifier))
```

Comment:
Oof, so, yeah, this is the nullifier privacy leakage we talked about. I'm still mulling over ways to reduce the degree of leakage.

Reply:
I think this is actually OK (though admittedly it's a big footgun). The wallet only actually uses this endpoint in two places
1. Right before it spends a nullifier
2. When it's checking if a transaction has completed

1. is fine because it means you're about to publish a transaction which includes this nullifier anyways, and it has presumably been some time since you received the spent record, so it's hard for an attacker to correlate this nullifier with an earlier record commitment.

2. is tricky. It's fine for the wallet that published the transaction to check it this way. The other wallet that might do this is the receiver checking on the status of a transaction they expect to receive. This is quite useful in testing but probably shouldn't be exposed in any actual product, because there's not really any private way to do it and outside of a testing situation where you know ahead of time the nullifiers of the transaction you're expecting to receive, it's not that useful.

Reply:
I love when markdown replaces "any number at all" with "1 + the last number I saw"